ghedesigner/ground_heat_exchangers.py

```diff
@@ -177,6 +177,77 @@ def _simulate_detailed(
 
         return hp_eft, delta_tb
 
+        # Source: Claesson, J., & Javed, S. (2012). A load-aggregation method to calculate extraction temperatures of
+        # borehole heat exchangers. ASHRAE Transactions, 118(1), 530-540.
+
+    def Simulate_Dynamic_LA(self, g, q_dot_hourly: np.ndarray):
+        rb = self.bhe.calc_effective_borehole_resistance()  # (m.K/W)
+        m_dot = self.bhe.m_flow_borehole  # (kg/s)
+        cp = self.bhe.fluid.cp  # (J/kg.s)
+
+        tg = self.bhe.soil.ugt  # (Celsius)
+        q_dot_b_Watt = np.hstack((q_dot_hourly / float(self.nbh)))  # W/m
+        height = self.bhe.b.H  # (meters)
+        q_dot = q_dot_b_Watt / height
+
+        two_pi_k = TWO_PI * self.bhe.soil.k  # (W/m.K)
+
+        n_max = len(q_dot)  # load_array_length
+        rate_expansion = 1.62
+        p_q = 9  # 5 # Number of bins in each level
+
+        q_max = int(np.log(n_max / p_q) / np.log(rate_expansion)) + 1  # Eq. 9
+        # v_max = p_q * (rate_expansion ** q - 1) / (rate_expansion - 1)
+
+        v_matrix = np.zeros((q_max, p_q + 1))  # Create v matrix
+
+        q_array = np.arange(1, q_max + 1)  # Creating levels
+        r_q_array = rate_expansion ** (q_array - 1)  # Rate of expansion
+
+        v_matrix[0, 0] = 0  # Eq. 10 top right
+
+        for i in range(1, q_max):  # Eq. 10 top left
+            v_matrix[i, 0] = v_matrix[i - 1, 0] + r_q_array[i - 1] * p_q
+
+        for q in range(0, q_max):  # Eq. 10 bottom
+            for p in range(1, p_q + 1):
+                v_matrix[q, p] = v_matrix[q, 0] + r_q_array[q] * p
+
+        ts = self.radial_numerical.t_s  # timescale
+
+        v_values_matrix = (v_matrix[:, 1:])  # Duration Matrix
+        v_array = v_values_matrix.flatten()  # Duration array in 1D
+        v_array = np.hstack((0.0, v_array))  # Adding beginning time
+        r_q_array_1d = np.zeros(q_max * p_q)  # Duration in each level in 1D
+
+        for i in range(0, len(r_q_array)):  # making 1d array of rq
+            r_q_array_1d[i * p_q:(i + 1) * p_q] = r_q_array[i]
+
+        hp_eft = np.zeros(len(q_dot))
+        delta_tb = np.zeros(len(q_dot))
+        g_s = g(np.log((v_array[1:] * 3600.0) / ts))  # g-values
+
+        Q_n = np.zeros(q_max * p_q)  # Array holding current load
```

Comment:
We're not totally consistent with Python naming conventions in GHEDesginer, but I would like to move that way. Following PEP 8, variable names should also be lower-cased. https://peps.python.org/pep-0008

Reply:
Modified the variables to lower case.